### PR TITLE
feat: Use Promise instead of future for start stop

### PR DESCRIPTION
**Pull request by samuel-hawker**
Switch to using Promise to remove casting future into promise
This is the vertx supported way of doing this v3.8<
https://github.com/vert-x3/wiki/wiki/3.8.0-Deprecations-and-breaking-changes#verticle-lifecycle

Signed-off-by: Samuel Hawker <samuel.hawker@ibm.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperator.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ public void start(Future<Void> start) {
 
     @Override
     @SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST")
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperator.java]: Do we still needs this annotation?

Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperator.java]: You are absolutely right, this is no longer needed. Removing now.

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@samuel-hawker Looks like the tests failed. Maybe the annotation was needed after all?

**Comment by samuel-hawker** [comment]
Very strange, it shouldn't be needed though...

I will do some local experimentation to determine why this is the case.

**Comment by samuel-hawker** [comment]
So I removed the suppression and not its import 🤦‍♂ Apologies!

**Comment by samuel-hawker** [comment]
Hadn't considered the other operators, just made the changes locally and testing now, will push up all other instances of where this change is needed shortly.

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=acceptance testcase=io.strimzi.systemtest.KafkaST#testLoadBalancerTls,io.strimzi.systemtest.ConnectST#testKafkaConnectWithFileSinkPlugin

**Pull request merged**